### PR TITLE
netbsd: Remove some env vars that are probably not needed.

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

Tool setup hooks will set most.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/bsd/netbsd/default.nix [hunk 1]
```diff
@@ -94,6 +86,9 @@ in lib.makeScopeWithSplicing
       platforms = platforms.unix;
       license = licenses.bsd2;
     };
+  } // lib.optionalAttrs stdenv'.hasCC {
+    # TODO should CC wrapper set this?
+    CPP = "${stdenv'.cc.targetPrefix}cpp";
```
Comment (alyssais) [inline pkgs/os-specific/bsd/netbsd/default.nix:91]: I'm worried that if we commit this we'll just forget about it and the TODO will be there forever.  How hard would it be to answer this question?

Comment (Ericson2314) [reply pkgs/os-specific/bsd/netbsd/default.nix:91]: @alyssais That's most of my TODOs, I'm sort of OK with that.

I think the issue is that clang didn't install a separate `cpp`, and lots of stuff does `gcc -E`/`clang -E`. I'm recalling when setting `LD` everywhere broke a bunch of things too.

Also, the wrapper might need to be adapted to assume the `-E` flag.

**Comment by alyssais** [comment]
Verified the NetBSD changes don't introduce any new pkgsCross.x86_64-netbsd.netbsd failures.

**Comment by alyssais** [comment]
NetBSD changes are good.
Don't think I'm really qualified to say anything about the stdenv changes other than that they seem to make sense from a stdenv user's PoV.

**Pull request merged**